### PR TITLE
read AHS data

### PR DESCRIPTION
**Pull request by maggiesu0725**

### CODE REVIEW

**Comment by maggiesu0725** [comment]
The program should be generating the number we want, but the number looks suspicious? Could you take a look? @lhendey @sstrochak

**Comment by maggiesu0725** [comment]
I used the WEIGHT variable as the weight,
"For Metro surveys, WEIGHT uses the boundaries as they were defined in the decennial year prior to the survey year. More information on weighting can be found at www.census.gov/programs-surveys/ahs/tech-documentation/def-errors-changes.html. WEIGHT may be used with all Metro surveys, and for National survey years 1997, 1999, and 2015 and later. For National survey years 2001 to 2013, WGT90GEO should be used as the final weight."

The average for SF:  2321, MF: 945

@lhendey

File: Macros/Apply AHS Formats.SAS [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+/* Specify location of value labels csv file.  INCLUDE the trailing backslash */
+%Let InDir = L:\Libraries\RegHsg\Raw\AHS 2017 Value Labels Package\;
+
+%LET CSVFile = AHS &Year. Value Labels.CSV;   /* name of input CSV file*/ 
+
+/* Specify location of AHS Public Use files. Include trailing backslash. Process metro and national files seperately from different file locations*/
+LIBNAME AHS "L:\Libraries\RegHsg\Raw\AHS 2017 Metropolitan PUF v1.0 SAS" ;
+
+/* location where you want the format catlog to be stored */
+LIBNAME Cat  "L:\Libraries\RegHsg\Raw\AHS 2017 Metropolitan PUF v1.0 SAS\" ;
```
Comment (lhendey) [inline Macros/Apply AHS Formats.SAS]: this should be to data not raw - that's probably why the other program isn't picking up the formats.

File: Prog/average_unitsize_sf_mf.sas [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ codebook: https://www.census.gov/data-tools/demo/codebook/ahs/ahsdict.html?s_appName=ahsdict&s_searchvalue=INTSTATUS
+ Modifications: 
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( RegHsg)
+
+/*
```
Comment (lhendey) [inline Prog/average_unitsize_sf_mf.sas]: if you fix the issue of where the formats are stored - i don't think you need these?

File: Prog/average_unitsize_sf_mf.sas [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+libname ahs "K:\Housefin3\Monthly\AHS";
+*/
+
+data test;
+set RegHsg.ahs2017m;
+keep OMB13CBSA;
+run;
+
+proc freq data=test;
+tables OMB13CBSA;
```
Comment (lhendey) [inline Prog/average_unitsize_sf_mf.sas]: i'm not seeing "47900"?

File: Prog/average_unitsize_sf_mf.sas [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+else if UNITSIZE="3" then area=875;
+else if UNITSIZE="4" then area=1250;
+else if UNITSIZE="5" then area=1750;
+else if UNITSIZE="6" then area=2250;
+else if UNITSIZE="7" then area=2750;
+else if UNITSIZE="8" then area=3500;
+else if UNITSIZE="9" then area=4000;
+run;
+
+proc tabulate data = DCMSA_2 missing;
```
Comment (lhendey) [inline Prog/average_unitsize_sf_mf.sas]: you'd need a table statement in proc tabulate. i'm not sure how you're generating the results (doesn't run for me)

File: Prog/average_unitsize_sf_mf.sas [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+set DCMSA;
+if UNITSIZE="1" then area=250;
+else if UNITSIZE="2" then area=625;
+else if UNITSIZE="3" then area=875;
+else if UNITSIZE="4" then area=1250;
+else if UNITSIZE="5" then area=1750;
+else if UNITSIZE="6" then area=2250;
+else if UNITSIZE="7" then area=2750;
+else if UNITSIZE="8" then area=3500;
+else if UNITSIZE="9" then area=4000;
```
Comment (lhendey) [inline Prog/average_unitsize_sf_mf.sas]: how does the unit size category end? 4000+?

**Review by lhendey — CHANGES_REQUESTED** [review]
@maggiesu0725 this isn't working for me.  see comments.

File: Prog/average_unitsize_sf_mf.sas (hunk 5 above)
Comment (maggiesu0725) [reply Prog/average_unitsize_sf_mf.sas]: yes @lhendey

File: Prog/average_unitsize_sf_mf.sas (hunk 3 above)
Comment (maggiesu0725) [reply Prog/average_unitsize_sf_mf.sas]: yes, 47900 dont exist in metropolitan dataset so you'll notice I used the national data in the actual calculation. @lhendey

**Review by lhendey — APPROVED** [review]

**Pull request merged**